### ISSUE TITLE
[Bug Report] Missing `.items()` in `HookedRootModule.hooks` 

### ISSUE THREAD

**Issue by dtch1997** [TransformerLensOrg/TransformerLens#611]
**Describe the bug**
Incorrect use of `self.hook_dict` in `HookedRootModule.hooks`  when iterating over backward hooks. 

**Code example**
```
def noop(act, hook):
    pass

with model.hooks(bwd_hooks = [(lambda x: True, noop)]): 
    pass
```

**Additional context**
https://github.com/TransformerLensOrg/TransformerLens/blame/5a374ec4b33cec6281b37494175d14f06c75dcfd/transformer_lens/hook_points.py#L306-L313

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

**Comment by bryce13950** [comment]
Thank you for opening this. We need to spend some time investigating this before we fix it. This file specifically has recently had quite a few changes in the pending 2.0 release. There is a possibility that the bug is fixed there, so a little bit of work needs to be done to see the full scope.

**Comment by neelnanda-io** [comment]
This actually is fixed in v2.0
https://github.com/TransformerLensOrg/TransformerLens/blame/295aca59ef73971e97bc7df5723f7f701de8497b/transformer_lens/hook_points.py#L374

**Comment by bryce13950** [comment]
I'll get it back ported into 1.x later today.

**Comment by bryce13950** [comment]
This has been resolved in 1.19, and a nice little test has been added to this to make sure this block is now covered for the future.
